libs/category/guides/architecture/filtering.md

```diff
@@ -0,0 +1,27 @@
+# Filter Architecture
+
+## Use Cases
+
+As a user, I would like to be able to apply filters to a collection of items of a category.
+As a user, I would like to know what filters are currently applied to the category. 
+As a user, I would like to see the filters applied IMMEDIATELY, without waiting for a [response from a backend platform (input-delay).](https://web.dev/fid/)
+
+## Features
+- [Filter Types](./filtering/types.md)
+  - [Equal](./filtering/types/equal.md)
+  - [Range](./filtering/types/range.md)
+- [Idempotence](./filtering/idempotence.md)
+
+## Implementation Details
+
+When an action is dispatched that would modify the "applied" filter state of the application, we intentionally optimistically apply the filters to the state of the category page. This means the following occurs in following order:
```

Comment:
why is "applied" in quotes?

Reply:
*intentionally, optimistically 

or

*intentionally and optimistically

I think is more correct

Reply:
"This means the following occurs in order:"

Reply:
8b2f50a1